modules/@apostrophecms/ui/ui/apos/components/AposContextMenu.vue

```diff
@@ -118,7 +118,8 @@ const dropdownContentStyle = ref({});
 const arrowEl = ref();
 
 defineExpose({
-  hide
+  hide,
+  setDropdownPosition
```

Comment:
The fact that you need to expose and reuse this method `setDropDownPosition` inside you implementation of `AposContext` is weird.
This should be run automatically when the context menu is opened.

From what I see in `seo-assistant` you try to manager the menu opening with `isModalVisible` and a `v-if`. I think this is where the mistake is. The menu opening should be an internal behavior.
Basically it generates the button, and when this button is clicked it opens the menu and compute the position.
It allows you to handle logic by providing some events:
* `@open`
* `@close`
* `@item-clicked`

Reply:
Menu is auto closed on cclick. This behaviour is deep into the component and it's children and can't be easily changed. We need to stay opened and show yet another modal. What we do is allow the slot content to be controlled from outside. Changing the slot content requires position recalculations, which is a more elegant solution.

Reply:
Nope, the general use case uses it internally to position the content initially and to listen for viewport changes (resize).